app/src/main/java/com/team41/boromi/BookActivity.java

```diff
@@ -328,4 +367,58 @@ public void onFailure(Exception e) {
   public void onBackPressed() {
 
   }
+
+  private void scanForBook() {
+    IntentIntegrator intentIntegrator = new IntentIntegrator(this);
+    intentIntegrator.setRequestCode(49374);
+    intentIntegrator.initiateScan();
+  }
+
+  private void queryBookData(String isbn) {
+    spinner.setVisibility(View.VISIBLE);
+    StringRequest stringRequest = new StringRequest(Request.Method.GET, formatQuery(isbn),
+        new Response.Listener<String>() {
+          @Override
+          public void onResponse(String response) {
+            // Display the first 500 characters of the response string.
+            displayBookData(response, isbn);
+          }
+        }, new Response.ErrorListener() {
+      @Override
+      public void onErrorResponse(VolleyError error) {
+        spinner.setVisibility(View.GONE);
+        makeToast("NETWORK ERROR OCCURRED ON REQUEST FOR BOOK");
+      }
+    });
+
+// Add the request to the RequestQueue.
+    rq.add(stringRequest);
+  }
+
+  private String formatQuery(String isbn) {
+    String formattedQuery = bookAPIURL;
+    formattedQuery += isbn;
+    formattedQuery += key;
+    return formattedQuery;
+  }
+
+  private void makeToast(String msg) {
+    Toast.makeText(this, msg, Toast.LENGTH_LONG).show();
+  }
+
+  private void displayBookData(String book, String isbn) {
+    System.out.println("DISPLAYING BOOKS");
+
+    GoogleBook gb = gson.fromJson(book, GoogleBook.class);
+    Book bookToDisplay = new Book();
+    bookToDisplay.setTitle(gb.getTitle(0));
+    bookToDisplay.setAuthor(gb.getFirstAuthor(0));
+    bookToDisplay.setISBN(isbn);
+    spinner.setVisibility(View.GONE);
+    DisplayBookFragment displayBookFragment = DisplayBookFragment.newInstance(bookToDisplay);
+    displayBookFragment.show(getSupportFragmentManager(), "displayUsers");
```

Comment:
"displayBooks" ?

Reply:
lmfaooo I stole this code from your method, works either way XD but I will change it

Reply:
lmfao you can leave it if it works

Reply:
nvm u changed it